internal/config/sidecar_test.go

```diff
@@ -55,57 +57,165 @@ func TestAgentSidecar_Bind(t *testing.T) {
-		           AutoBackupDuration: "",
-		           PostStopTimeout: "",
-		           Filename: "",
-		           FilenameSuffix: "",
-		           BlobStorage: Blob{},
-		           Compress: CompressCore{},
-		           RestoreBackoff: Backoff{},
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
+			mode := "sidecar"
+			watchDir := "sidecar"
+			autoBackupDuration := "10ms"
+			postStopTimeout := "5m"
+			filename := "vald-ngt-1"
+			filenameSuffix := "tar.gz"
+			blobStorageType := "s3"
+			compressAlgorithm := GOB.String()
+			backoffInitialDuration := "10ms"
+			return test{
+				name: "return AgentSidecar when all of object are set",
+				fields: fields{
+					Mode:               mode,
+					WatchDir:           watchDir,
+					AutoBackupDuration: autoBackupDuration,
+					PostStopTimeout:    postStopTimeout,
+					Filename:           filename,
+					FilenameSuffix:     filenameSuffix,
+					BlobStorage: &Blob{
+						StorageType: blobStorageType,
+					},
+					Compress: &CompressCore{
+						CompressAlgorithm: compressAlgorithm,
+					},
+					RestoreBackoff: &Backoff{
+						InitialDuration: backoffInitialDuration,
+					},
+					Client: &Client{
+						Net: new(Net),
+					},
+				},
+				want: want{
+					want: &AgentSidecar{
+						Mode:               mode,
+						WatchDir:           watchDir,
+						AutoBackupDuration: autoBackupDuration,
+						PostStopTimeout:    postStopTimeout,
+						Filename:           filename,
+						FilenameSuffix:     filenameSuffix,
+						BlobStorage: &Blob{
+							StorageType: blobStorageType,
+							S3:          new(S3Config),
+						},
+						Compress: &CompressCore{
+							CompressAlgorithm: compressAlgorithm,
+						},
+						RestoreBackoff: &Backoff{
+							InitialDuration: backoffInitialDuration,
+						},
+						Client: &Client{
+							Net: new(Net),
+						},
+					},
+				},
+			}
+		}(),
+		func() test {
+			mode := "sidecar"
+			watchDir := "sidecar"
+			autoBackupDuration := "10ms"
+			postStopTimeout := "5m"
+			filename := "vald-ngt-1"
+			filenameSuffix := "tar.gz"
+			return test{
+				name: "return AgentSidecar when all of object are not set",
+				fields: fields{
+					Mode:               mode,
+					WatchDir:           watchDir,
+					AutoBackupDuration: autoBackupDuration,
+					PostStopTimeout:    postStopTimeout,
+					Filename:           filename,
+					FilenameSuffix:     filenameSuffix,
+				},
+				want: want{
+					want: &AgentSidecar{
+						Mode:               mode,
+						WatchDir:           watchDir,
+						AutoBackupDuration: autoBackupDuration,
+						PostStopTimeout:    postStopTimeout,
+						Filename:           filename,
+						FilenameSuffix:     filenameSuffix,
+						BlobStorage:        new(Blob),
+						Compress:           new(CompressCore),
+						RestoreBackoff:     new(Backoff),
+						Client:             new(Client),
+					},
+				},
+			}
+		}(),
```

Comment:
seems not much between test name and the fields parameter

Reply:
thank you. I fixed.